earn/src/components/borrow/modal/AddCollateralModal.tsx

```diff
@@ -131,142 +26,65 @@ export type AddCollateralModalProps = {
+    if (uniswapNFTPositions.size > 0) {
+      return AddCollateralModalState.SELECT_COLLATERAL_TYPE;
+    }
+    return AddCollateralModalState.TOKENS;
   });
 
   useEffect(() => {
-    let interval: NodeJS.Timer | null = null;
     if (isOpen) {
-      interval = setInterval(() => {
-        refetchBalance();
-      }, 13_000);
-    }
-    if (!isOpen && interval != null) {
-      clearInterval(interval);
+      setModalState(() => {
+        if (uniswapNFTPositions.size > 0) {
+          return AddCollateralModalState.SELECT_COLLATERAL_TYPE;
+        }
+        return AddCollateralModalState.TOKENS;
+      });
     }
-    return () => {
-      if (interval != null) {
-        clearInterval(interval);
-      }
-    };
-  }, [isOpen, refetchBalance]);
-
-  // Reset collateral amount and token when modal is opened/closed or when the margin account token0 changes
-  useEffect(() => {
-    setCollateralAmount('');
-    setCollateralToken(marginAccount.token0);
-  }, [isOpen, marginAccount.token0]);
+  }, [isOpen, uniswapNFTPositions.size]);
 
-  const tokenOptions = [marginAccount.token0, marginAccount.token1];
-
-  const existingCollateral =
-    collateralToken.address === marginAccount.token0.address
-      ? marginAccount.assets.token0Raw
-      : marginAccount.assets.token1Raw;
-
-  const numericCollateralAmount = Number(collateralAmount) || 0;
-
-  const collateralAmountBig = new Big(numericCollateralAmount).mul(10 ** collateralToken.decimals);
-
-  const existingCollateralBig = new Big(existingCollateral).mul(10 ** collateralToken.decimals);
-
-  const numericUserBalance = Number(userBalance?.formatted ?? 0) || 0;
-  const userBalanceBig = new Big(numericUserBalance).mul(10 ** collateralToken.decimals);
-
-  if (!userAddress || !isOpen) {
-    return null;
-  }
-
-  const newCollateralBig = existingCollateralBig.add(collateralAmountBig).div(10 ** collateralToken.decimals);
+  const defaultUniswapNFTPosition = uniswapNFTPositions.size > 0 ? Array.from(uniswapNFTPositions.entries())[0] : null;
 
   return (
-    <Modal isOpen={isOpen} title='Add Collateral' setIsOpen={setIsOpen} maxHeight='650px'>
-      <div className='flex flex-col items-center justify-center gap-8 w-full mt-2'>
-        <div className='flex flex-col gap-1 w-full'>
-          <div className='flex flex-row justify-between mb-1'>
-            <Text size='M' weight='bold'>
-              Collateral Amount
-            </Text>
-            <BaseMaxButton
-              size='L'
-              onClick={() => {
-                if (userBalance != null) {
-                  setCollateralAmount(userBalance?.formatted);
-                }
-              }}
+    <Modal isOpen={isOpen} title='Add Collateral' setIsOpen={setIsOpen} maxHeight='650px' maxWidth='500px'>
+      {modalState === AddCollateralModalState.SELECT_COLLATERAL_TYPE && (
+        <div className='flex flex-col gap-4'>
+          <Text size='L'>What type of collateral would you like to add?</Text>
```

Comment:
I think this text should be smaller and/or a different color than the modal color.